lib/Dialect/FIRRTL/LowerToRTL.cpp

```diff
@@ -440,6 +461,39 @@ LogicalResult FIRRTLLowering::lowerBinOp(Operation *op) {
   return setLoweringTo<ResultOpType>(op, lhs, rhs);
 }
 
+/// lowerCmpOp extends each operand to the longest type, then performs the
+/// specified binary operator.
+template <typename ResultOpSignedType, typename ResultOpUnsignedType>
+LogicalResult FIRRTLLowering::lowerCmpOp(Operation *op, bool flip) {
+  // Extend the two operands to match the longest type.
+  Type resultType = builder->getIntegerType(1);
+  auto lhsFIRType =
+      op->getOperand(0).getType().cast<FIRRTLType>().getPassiveType();
+  auto lhsIntType = lhsFIRType.dyn_cast<IntType>();
+  auto rhsFIRType =
+      op->getOperand(1).getType().cast<FIRRTLType>().getPassiveType();
+  auto rhsIntType = rhsFIRType.dyn_cast<IntType>();
+  
+  if (!lhsIntType || !lhsIntType.hasWidth() || !rhsIntType || !rhsIntType.hasWidth())
```

Comment:
This defensive and yucky casting logic makes me sad, but I agree it is necessary right now.  I filed [issue 140](https://github.com/llvm/circt/issues/140) to track fixing this someday.